src/encord_active/analysis/metrics/image/area.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, cast
+
+import torch
+
+from encord_active.analysis.metric import (
+    ImageObjectOnlyOutputBatch,
+    MetricDependencies,
+    ObjectOnlyBatchInput,
+    OneImageMetric,
+)
+from encord_active.analysis.types import (
+    BoundingBoxTensor,
+    ImageBatchTensor,
+    ImageTensor,
+    MaskTensor,
+    MetricBatchDependencies,
+    MetricResult,
+)
+from encord_active.analysis.util import image_height, image_width
+from encord_active.analysis.util.torch import batch_size
+from encord_active.db.metrics import MetricType
+
+
+class AreaMetric(OneImageMetric):
+    def __init__(self) -> None:
+        super().__init__(
+            ident="metric_area",
+            long_name="Area",
+            desc="Area in pixels",
+            metric_type=MetricType.UINT,
+        )
+
+    def calculate(
+        self, deps: MetricDependencies, image: ImageTensor, mask: Optional[MaskTensor], bb: Optional[BoundingBoxTensor]
+    ) -> MetricResult:
+        if mask is None:
+            return float(image_width(image)) * float(image_height(image))
+        else:
+            return cast(torch.Tensor, deps["ephemeral_grayscale_image"]).shape[0]
```

Comment:
Can we just remove `cast` from all over the code as it seems a bit dangerous, instead what about checking the type with `isinstance()`, like:

```
if isinstance(deps["ephemeral_grayscale_image"], torch.Tensor):
     return deps["ephemeral_grayscale_image"].shape[0]
 else:
    raise TypeError(f"Expected torch.Tensor, got {type(deps["ephemeral_grayscale_image"])}")
```

Reply:
yes - though easiest is probably to use a typed dict for the dependency dict.